File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String res = "";
        String tmp = "";
        s = s+ " ";
        int cntSpace = 0;
        for(int i=0;i<s.length();i++){
            
            char ch = s.charAt(i);
            if(ch == ' '){
                cntSpace++;
                if(cntSpace <=1){
                    res = tmp+ " " + res ; 
                    tmp = "";
                }
                
            }else{
                cntSpace = 0;
                tmp += ch;
            }
        }
        res = res.trim();
        return res;
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        st = ""
        x = list(s.split())
        print(x[0])
        
        for i in range(len(x)):
            st = x[i]+" " + st 
          
        return st[:-1];    
        